spec-dtslint/operators/tap-spec.ts

```diff
@@ -0,0 +1,20 @@
+import { of } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+it('should infer correctly', () => {
+  const a = of(1, 2, 3).pipe(tap()); // $ExpectType Observable<number>
+});
+
+it('should accept partial observer', () => {
+  const a = of(1, 2, 3).pipe(tap({ next: (x: number) => x })); // $ExpectType Observable<number>
```

Comment:
Don't return a value from the `next` handler. TypeScript is okay with this, but the test should reflect the intended usage.